newIDE/app/src/Profile/Subscription/PlanCard.js

```diff
@@ -14,59 +18,197 @@ import { useResponsiveWindowWidth } from '../../UI/Reponsive/ResponsiveWindowMea
 import Silver from './Icons/Silver';
 import Gold from './Icons/Gold';
 import GDevelopGLogo from '../../UI/CustomSvgIcons/GDevelopGLogo';
+import { selectMessageByLocale } from '../../Utils/i18n/MessageByLocale';
 
 const styles = {
   bulletIcon: { width: 20, height: 20, marginRight: 10 },
   bulletText: { flex: 1 },
 };
 
-const getPlanPrice = ({
-  plan,
+const getPlanPrice = (price: SubscriptionPlanPrice): React.Node => {
+  if (price.period === 'week') {
+    if (price.periodCount === 1) {
+      if (price.isPerUser) {
+        return (
+          <Text key="week" noMargin color="secondary">
+            <Trans>
+              {price.amountInCents / 100}
```

Comment:
this feels repeated everywhere, though it's slightly different every time so unsure it can be better

Reply:
maybe at least 1 function `formatPriceAndCurrency` ?

Reply:
Good idea, I'll add it!